### PR TITLE
Updated the version Python to 3.9.18 as needed for OpenSSL Windows update

### PR DESCRIPTION
**Pull request by carlosroman**
### What does this PR do?

Upgrades the version of OpenSSL being used to `1.1.1w`. Fixes CVE-2023-4807, CVE-2023-3817 and CVE-2023-3446.
See https://www.openssl.org/news/openssl-1.1.1-notes.html for more details.

Windows OpenSSL upgrade: [OpenSSL 1.1.1w](https://github.com/DataDog/cpython/pull/29).

### Motivation

The current version of OpenSSL being used on Windows (version `1.1.1u`) has some reported CVEs that are mitigated by version `1.1.1w`.

### Describe how to test/QA your changes

* Run an Agent with Python 3 (either Agent 6 configured with py3 or an Agent 7)
* Execute agent status and validate Python version
* Validate the Python integrations are working properly

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by chouquette — APPROVED** [review]

File: omnibus/config/software/python3.rb [hunk 1]
```diff
@@ -58,19 +58,19 @@
...
 
 else
-  default_version "3.9.17-26e6052"
+  default_version "3.9.18-38f3b72"
   dependency "vc_redist_14"
 
   if windows_arch_i386?
     dependency "vc_ucrt_redist"
 
     source :url => "https://dd-agent-omnibus.s3.amazonaws.com/python-windows-#{version}-x86.zip",
```
[the remark this answers is not in the document]
Comment (carlosroman) [reply omnibus/config/software/python3.rb:67]: This is done via this gitlab pipeline defined [here](https://github.com/DataDog/cpython/blob/3.9.x/.gitlab-ci.yml#L49).

[the remark this answers is not in the document]
Comment (carlosroman) [reply omnibus/config/software/python3.rb:67]: When we move to Python 3.11 we will no longer be using that fork of Python.

File: releasenotes/notes/update-python-openssl-on-windows-to-1.1.1w-0a00d9c8a0cfde12.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#
+# Each section note must be formatted as reStructuredText.
+---
+upgrade:
+  - |
+    Upgrade Python 3.9 to Python 3.9.18
+
+security:
+  - |
+    Updated the version of OpenSSL used by Python on Windows to `1.1.1w`; addressed CVE-2023-4807, CVE-2023-3817 and CVE-2023-3446.
```
Comment (buraizu) [inline releasenotes/notes/update-python-openssl-on-windows-to-1.1.1w-0a00d9c8a0cfde12.yaml:15]:
```suggestion
    Updated the version of OpenSSL used by Python on Windows to `1.1.1w`; addressed CVE-2023-4807, CVE-2023-3817, and CVE-2023-3446
```

File: releasenotes/notes/update-python-openssl-on-windows-to-1.1.1w-0a00d9c8a0cfde12.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+upgrade:
+  - |
+    Upgrade Python 3.9 to Python 3.9.18
```
Comment (buraizu) [inline releasenotes/notes/update-python-openssl-on-windows-to-1.1.1w-0a00d9c8a0cfde12.yaml:11]:
```suggestion
    Upgraded Python 3.9 to Python 3.9.18
```

**Review by buraizu — APPROVED** [review]
Approving with some very minor updates for consistency

**Pull request merged**